src/ngAnimate/animateCss.js

```diff
@@ -588,17 +590,10 @@ var $AnimateCssProvider = ['$animateProvider', function($animateProvider) {
         blockTransitions(node, SAFE_FAST_FORWARD_DURATION_VALUE);
       }
 
-      var timings = computeTimings(node, fullClassName, cacheKey);
-      var relativeDelay = timings.maxDelay;
-      maxDelay = Math.max(relativeDelay, 0);
-      maxDuration = timings.maxDuration;
-
       var flags = {};
       flags.hasTransitions          = timings.transitionDuration > 0;
       flags.hasAnimations           = timings.animationDuration > 0;
-      flags.hasTransitionAll        = flags.hasTransitions && timings.transitionProperty == 'all';
-      flags.applyTransitionDuration = hasToStyles && (
-                                        (flags.hasTransitions && !flags.hasTransitionAll)
+      flags.applyTransitionDuration = options.duration > 0 || hasToStyles && (flags.hasTransitions
                                          || (flags.hasAnimations && !flags.hasTransitions));
```

Comment:
I am not sure if this expressionis correct, but if it is, it can be simplified to:

`options.duration > 0 || hasToStyles && (flags.hasTransitions || flags.hasAnimations)`